sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2616,4 +2617,66 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
     val e = intercept[AnalysisException](sql("SELECT nvl(1, 2, 3)"))
     assert(e.message.contains("Invalid number of arguments"))
   }
+
+  test("TRIM function-BOTH") {
+    val ae1 = intercept[ParseException]{
+      sql("select lower(BOTH 'S' FROM 'SS abc S')").head
```

Comment:
Why test `lower`? Is this related to this pr? I think we should test other trim option names instead of other funtions